File: backend/app/schemas/trust.py
from datetime import datetime
from enum import Enum
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, ConfigDict, Field


class RiskLevel(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    CRITICAL = "critical"
    UNKNOWN = "unknown"


class TrustScoreResponse(BaseModel):
    id: UUID
    company_id: UUID
    research_run_id: UUID
    score: Optional[float] = Field(default=None, ge=0.0, le=100.0)
    confidence: Optional[float] = Field(default=None, ge=0.0, le=1.0)
    risk_level: Optional[RiskLevel] = RiskLevel.UNKNOWN
    evidence_coverage: Optional[float] = Field(default=None, ge=0.0, le=1.0)
    algorithm_version: str = "v1.0"
    explanation: Optional[str] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/trust.ts
import type { VerificationStatus } from './evidence';

export type RiskLevel = 'low' | 'medium' | 'high' | 'critical' | 'unknown';

export interface TrustScore {
  id?: string;
  company_id: string;
  research_run_id: string;
  score: number;
  confidence: number;
  risk_level: RiskLevel;
  evidence_coverage: number;
  algorithm_version: string;
  explanation?: string | null;
  created_at?: string;
}

export interface TrustSignalBreakdown {
  dimension: string;
  label: string;
  score: number;
  maxScore: number;
  weight: number;
  status: VerificationStatus;
  description: string;
}
